Src/FluentAssertions.Json/JTokenAssertions.cs

```diff
@@ -180,6 +181,59 @@ public AndConstraint<JTokenAssertions> NotHaveValue(string unexpected, string be
             return new AndConstraint<JTokenAssertions>(this);
         }
 
+        /// <summary>
+        ///     Asserts that the current <see cref="JToken" /> match the specified <paramref name="regularExpression" /> regular expression pattern.
+        /// </summary>
+        /// <param name="regularExpression">The expected regular expression pattern</param>
+        public AndConstraint<JTokenAssertions> MatchRegex(string regularExpression)
+        {
+            return MatchRegex(regularExpression, string.Empty);
+        }
+
+        /// <summary>
+        ///     Asserts that the current <see cref="JToken" /> match the specified <paramref name="regularExpression" /> regular expression pattern.
+        /// </summary>
+        /// <param name="regularExpression">The expected regular expression pattern</param>
+        /// <param name="because">
+        ///     A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        ///     is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        ///     Zero or more objects to format using the placeholders in <see paramref="because" />.
+        /// </param>
+        public AndConstraint<JTokenAssertions> MatchRegex(string regularExpression, string because, params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .ForCondition(Regex.IsMatch(Subject.Value<string>(), regularExpression))
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected JSON property {0} to have regex pattern {1}{reason}, but found {2}.",
```

Comment:
🔧  Instead of `JSON property`, use `{context:JSON property}` instead. If somebody wraps this call in `AssertionScope`, it will allow them to use something more specific than just `JSON property`
🔧 I _think_ you mean _Expected.....to match regex pattern_, right?